File: 프로그래머스/Lv1/가운데 글자 가져오기.py
# 내 풀이
def solution(s):
    answer = ""
    
    if len(s) % 2 == 0 : 
        answer = s[len(s)//2-1:len(s)//2+1] 
    else : 
        answer = s[len(s)//2]
        
    return answer

def solution2(s):
    return str[(len(s)-1)//2 : len(s)//2 + 1]

print(solution("abcde"))
print(solution2("qwer"))

File: 프로그래머스/Lv1/가운데 글자 가져오기.js
// 내 풀이
function solution(s) {
    let i = s.length / 2;
    //str.substring(시작 인덱스번호, 끝 인덱스번호)
    return s.length % 2 ? s[Math.floor(i)] : s.substring(i-1, i+1); 
}

// 다른 사람 풀이
function solution2(s) {
    // str.substr(시작 인덱스, 추출 갯수)
    return s.substr(Math.ceil(s.length/2)-1, s.length%2 ? 1 : 2);
}

console.log(solution("abcde"));
console.log(solution2("qwer"));